src/modules/modal.ts

```diff
@@ -0,0 +1,51 @@
+const SHOW = 'modal/SHOW' as const;
+const HIDE = 'modal/HIDE' as const;
+const CHANGE_MODAL = 'modal/CHANGE_MODAL' as const;
+export enum ModalTypes {
+  none = 'none',
+  LogIn = 'LogIn',
+  SignIn = 'SignIn',
+  BugReport = 'BugReport',
+  ChangePW = 'ChangePW',
+}
+export const showModal = () => ({
+  type: SHOW,
+});
+export const hideModal = () => ({
+  type: HIDE,
+});
+
+export const changeModal = (modalType: ModalTypes) => ({
+  type: CHANGE_MODAL,
+  payload: modalType,
+});
+
+type ModalAction =
+  | ReturnType<typeof showModal>
+  | ReturnType<typeof hideModal>
+  | ReturnType<typeof changeModal>;
+
+export interface ModalState {
+  isVisible: boolean;
+  modalType: ModalTypes;
+}
+
+const initialState: ModalState = {
+  isVisible: false,
+  modalType: ModalTypes.none,
+};
+
+function modal(state = initialState, action: ModalAction): ModalState {
+  switch (action.type) {
+    case SHOW:
+      return { ...state, isVisible: true };
+    case HIDE:
+      return { ...state, isVisible: false };
```

Comment:
```suggestion
    case TOGGLE:
        return { ...state, isVisible: action.payload.isVisible }
```
굳이 true false를 toggle하는 이벤트를 액션 두개로 나눌 필요가 있나 싶습니다. 
액션 dispatch 호출할 때 인자로 boolean 값을 넘기거나 `!isVisible` 이 부분에 대해서 고려해보는 것은 어떤가 싶네요.

Reply:
넵 그렇네요 하나의 엑션으로 관리하는쪽으로 수정하겟습니다